File: backend/dataset/levelOrder/levelorder_12.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        
        
        traversal_queue = [ root ] if root else []
        
        path = []
        
        while traversal_queue:
            
            cur_level_path, next_level_queue = [], []
            
            for node in traversal_queue:
                
                # update current level traversal path
                cur_level_path.append( node.val )
                
                if node.left:
                    next_level_queue.append( node.left )
                
                if node.right:
                    next_level_queue.append( node.right )
            
            # add current level path into path collection
            path.append( cur_level_path )
            
            # update next_level_queue as traversal_queu
            traversal_queue = next_level_queue
            
        return path

File: backend/CDataset/levelOrder/levelOrder_12.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    class TreeNode{
        public:
        TreeNode* left;
        TreeNode* right;
        int val;
    };

    vector<vector<int>> ans;
    
    void dfs(TreeNode* root, int level) {
        if (!root) 
            return;
        
        if (level == ans.size()) {
            ans.push_back({});
        }
        
        ans[level].push_back(root->val);
        
        dfs(root->left, level+1);
        dfs(root->right, level+1);
    }
    
    vector<vector<int>> levelOrder(TreeNode* root) {
        dfs(root, 0);
        
        return ans;
    }
};
